add a lua script to include short titles on headers (#310)
rstudio/rticles: inst/rmarkdown/templates/jss_article/skeleton/skeleton.Rmd

author:
  - name: FirstName LastName
    affiliation: University/Company
    address: |
      | First line

>       | Second line

Reviewer: Are the leading pipes necessary? Probably not. Could you [double check](https://pandoc.org/MANUAL.html#metadata-blocks)?

Author: I blindly commit the suggestion  in #254, but I'll check that.

Reviewer: I just checked it. They are necessary. The meanings of the pipe after and below `address:` are different. The former is explained at https://pandoc.org/MANUAL.html#metadata-blocks and the latter means https://pandoc.org/MANUAL.html#line-blocks

Author: Thanks ! That is good to know!

One exception is inline code, which can be written inside a pair of backticks (i.e., using the Markdown syntax).

> If you want to use LaTex commands in headers, you need to provide a `short-title`

Reviewer: ```suggestion
If you want to use LaTeX commands in headers, you need to provide a `short-title`
```

attribute. You can also provide a custom identifier if necessary.

> See the example of the header \ref{r-code}

Reviewer: ```suggestion
See the header of Section \ref{r-code} for example.
```

BTW, since other paragraphs are not wrapped, let's not wrap this one, either.

If you prefer hard wrapping, we can use the RStudio daily build with the visual editor, so you don't have to manually wrap the text.

# \proglang{R} code {short-title="R code" #r-code}

Can be inserted in regular R markdown blocks.

```{r}
x <- 1:10
x
```

## Features specific to \pkg{rticles} {short-title="Features specific to rticles"}

> * Adding short title to section is specific to \pkg{rticles} as currently unsupported by pandoc. It is powered by a pandoc lua filter.

Reviewer: ```suggestion
* Adding short titles to section headers is a feature specific to \pkg{rticles} (implemented via a Pandoc Lua filter). This feature is currently not supported by Pandoc and we will update this template if [it is officially supported in the future](https://github.com/jgm/pandoc/issues/4409).
```

> * Using `\AND` syntax in author field to add author in a new line. This is a specific `jss_article` format.

Reviewer: ```suggestion
* Using the `\AND` syntax in the `author` field to add authors on a new line. This is a specific to the `rticles::jss_article` format.
```
